dags/keyword_extraction_pipeline.py

```diff
@@ -57,94 +47,148 @@ def get_default_args():
+
+
 def get_data_config(**kwargs):
-    conf_file_path = get_env_var_or_use_default(
-        PEERSCOUT_CONFIG_FILE_PATH_ENV_NAME, None
+    conf_file_path = os.getenv(
+        EXTRACT_KEYWORDS_CONFIG_FILE_PATH_ENV_NAME
+    )
+    data_config_dict = get_yaml_file_as_dict(
+        conf_file_path
+    )
+    kwargs["ti"].xcom_push(
+        key="multi_keyword_extract_conf_dict",
+        value=data_config_dict
     )
-    data_config_dict = copy.deepcopy(DEFAULT_CONFIG)
-    if conf_file_path:
-        data_config_dict.update(
-            get_yaml_file_as_dict(conf_file_path)
-        )
-    kwargs["ti"].xcom_push(key="data_config_dict",
-                           value=data_config_dict)
 
 
 def etl_extraction_keyword(**kwargs):
     dag_context = kwargs["ti"]
-    data_config_dict = dag_context.xcom_pull(
-        key="data_config_dict", task_ids="get_data_config"
+    multi_keyword_extract_conf_dict = dag_context.xcom_pull(
+        key="multi_keyword_extract_conf_dict", task_ids="get_data_config"
     )
     # handles the external triggers
     externally_triggered_parameters = kwargs['dag_run'].conf or {}
     limit_row_count_value = externally_triggered_parameters.get(
         ExternalTriggerConfig.LIMIT_ROW_COUNT
     )
-
     dep_env = (
         externally_triggered_parameters.get(
-            ExternalTriggerConfig.BQ_DATASET_PARAM_KEY,
-            get_env_var_or_use_default(DEPLOYMENT_ENV,
-                                       DEFAULT_DEPLOYMENT_ENV_VALUE)
+            ExternalTriggerConfig.DEPLOYMENT_ENV,
+            os.getenv(
+                DEPLOYMENT_ENV
+            )
         )
-
     )
     table = externally_triggered_parameters.get(
         ExternalTriggerConfig.BQ_TABLE_PARAM_KEY
     )
     spacy_language_model = externally_triggered_parameters.get(
         ExternalTriggerConfig.SPACY_LANGUAGE_MODEL_NAME_KEY
     )
+    multi_keyword_extract_conf = MultiKeywordExtractConfig(
+        multi_keyword_extract_conf_dict,
+        dep_env
+    )
+    state_dict = get_stored_state(
+        multi_keyword_extract_conf.state_file_bucket_name,
+        multi_keyword_extract_conf.state_file_object_name
+    )
+    timestamp_as_string = current_timestamp_as_string()
+    for extract_conf_dict in multi_keyword_extract_conf.keyword_extract_config:
+        with NamedTemporaryFile() as named_temp_file:
+            keyword_extract_config = KeywordExtractConfig(
+                extract_conf_dict,
+                gcp_project=multi_keyword_extract_conf.gcp_project,
+                destination_table=table,
+                limit_count_value=limit_row_count_value,
+                spacy_language_model=spacy_language_model,
+                import_timestamp_field_name=(
+                    multi_keyword_extract_conf.import_timestamp_field_name
+                )
+            )
+            etl_and_update_state(
+                keyword_extract_config,
+                state_dict,
+                named_temp_file.name,
```

Comment:
minor: I think it is not clear what the temporary file is for? Maybe we shouldn't need to pass it in, if it's not used. Was there a reason not to just create a temporary file where it seems to be needed, I believe `etl_keywords_get_latest_state`?

Reply:
yeah , you are right, i used to use the file name in this function before but not anymore, so , yes, no need for it